pkg/kubectl/cmd/attach.go

```diff
@@ -167,7 +167,7 @@ func (p *AttachOptions) Run() error {
 		if err != nil {
 			return err
 		}
-		if pod.Status.Phase != api.PodRunning {
+		if pod.Status.Phase != api.PodRunning && (pod.Status.Phase == api.PodPending && len(pod.Spec.InitContainers) < 1) {
```

Comment:
use `len(pod.Spec.InitContainers) == 0)` instead of `len(pod.Spec.InitContainers) < 1)`?